### PR TITLE
fix(select): add warning for value prop miss-use

### PR DESCRIPTION
**Pull request by gergelyke**
Closes https://github.com/uber/baseweb/issues/2586

### CODE REVIEW

File: src/select/select-component.js [hunk 1]
```diff
@@ -849,6 +849,13 @@ class Select extends React.Component<PropsT, SelectStateT> {
       value,
       filterOutSelected,
     } = this.props;
+
+    if (__DEV__ && !Array.isArray(value)) {
```
Comment (chasestarr) [inline src/select/select-component.js]: the dev check needs to be separate from other conditionals

Comment (gergelyke) [reply src/select/select-component.js]: okay, I'll make the changes - can you help me why they need to be separated?

Comment (chasestarr) [reply src/select/select-component.js]: the code won't be removed from prod builds

Comment (gergelyke) [reply src/select/select-component.js]: thank you Chase - just updated the PR

**Pull request merged**